### PR TITLE
modify ghost logic

### PR DESCRIPTION
**Pull request by javadocq**
Changed the code logic for the ghost item feature. It used to use threads, but the new logic uses cooldown classes.

Also, the color change was done in a thread, whereas the new logic uses the GREEN -> Dark.Gray 
Then, in GameScreen.update, we changed it to DarkGray -> GREEN.

### CODE REVIEW

File: src/engine/ItemManager.java [hunk 1]
```diff
@@ -48,6 +48,10 @@ public class ItemManager {
     private boolean timeStopActive;
     /** Check if Ghost is active */
     private boolean ghostActive;
+    /** Cooldown of Ghost */
+    private static final int GHOST_COOLDOWN = 3000;
```
Comment (dev-moonsc) [inline src/engine/ItemManager.java:52]: It is constant value. so, How about moving this code to Line 35?

File: src/engine/ItemManager.java [hunk 2]
```diff
@@ -81,6 +85,7 @@ public ItemManager(Ship ship, EnemyShipFormation enemyShipFormation, Set<Barrier
         this.enemyShipFormation = enemyShipFormation;
         this.barriers = barriers;
         this.logger = Core.getLogger();
+        this.ghostActive = false;
```
Comment (dev-moonsc) [inline src/engine/ItemManager.java:88]: The boolean type variable is automatically initialized false when it is declared. So, I think this line is unnecessary.

**Review by dev-moonsc — CHANGES_REQUESTED** [review]
LGTM :) But, please check the chore changes that I requested.

File: src/engine/ItemManager.java [hunk 3]
```diff
@@ -264,18 +269,9 @@ private Entry<Integer, Integer> operateBarrier() {
...
-            try {
-                Thread.sleep(3000);
-                this.ghostActive = false;
-                this.ship.setColor(Color.GREEN);
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            }
-        }).start();
+        ghost_cooldown = Core.getCooldown(GHOST_COOLDOWN);
+        ghost_cooldown.reset();
```
Comment (dev-moonsc) [inline src/engine/ItemManager.java:274]: You delete line 269, the sound effect of Ghost. Please resolve this conflict.

**Review by dev-moonsc — CHANGES_REQUESTED** [review]

File: src/engine/ItemManager.java (hunk 3 above)
Comment (javadocq) [reply src/engine/ItemManager.java:274]: I mistake to delete this line.
I resolve that

**Pull request closed without merging**